src/optimizer/input_column_deriver.cpp

```diff
@@ -219,6 +219,8 @@ void InputColumnDeriver::AggregateHelper(const BaseOperatorNode *op) {
   // Check having predicate, since the predicate may use columns other than
   // output columns
   for (auto &having_expr : having_exprs) {
+    // We perform aggregate here so the output contains aggregate exprs while
+    // input should contain all tuple value exprs used to perform aggregation.
     expression::ExpressionUtil::GetTupleValueExprs(input_cols_set,
                                                    having_expr.expr.get());
     expression::ExpressionUtil::GetTupleAndAggregateExprs(
```

Comment:
Then here why we're using `GetTupleAndAggregateExprs`? Shouldn't we only get aggregate exprs?

Reply:
What's the answer for this?

Reply:
I missed this earlier. If we have a query `select avg(a), b from test group by b having sum(c) + b < 10`, then the output should contain both `sum(c)` and `b`.

Reply:
But what if we have a query `select avg(a), b from test group by b having sum(c) < 10`? Wouldn't the output have both `sum(c)` and `c` according to your rule?

Reply:
It would not get `c`. It only gets TVEs without aggregate.

Reply:
I see... I think this answers all my questions!